src/hotspot/share/cds/dynamicArchive.hpp

```diff
@@ -25,18 +25,18 @@
 #ifndef SHARE_VM_MEMORY_DYNAMICARCHIVE_HPP
 #define SHARE_VM_MEMORY_DYNAMICARCHIVE_HPP
 
-#if INCLUDE_CDS
-
-#include "classfile/compactHashtable.hpp"
+#include "cds/compactHashtable.hpp"
+#include "cds/filemap.hpp"
 #include "memory/allocation.hpp"
-#include "memory/filemap.hpp"
 #include "memory/memRegion.hpp"
 #include "memory/virtualspace.hpp"
 #include "oops/oop.hpp"
 #include "utilities/exceptions.hpp"
 #include "utilities/macros.hpp"
 #include "utilities/resourceHash.hpp"
 
+#if INCLUDE_CDS
```

Comment:
I have to wonder who is including this file and why, if CDS is not enabled.

Reply:
E.g., jvm.cpp includes dynamicArchive.hpp, but only uses its declarations inside INCLUDE_CDS blocks. It would be too verbose to do this in jvm.cpp

```
#if INCLUDE_CDS
#include "cds/dynamicArchive.hpp"
#endif
```

Reply:
We routinely do that for other INCLUDE_X features.